app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,53 @@
+class CommentsController < ApplicationController
+  include ApplicationHelper
+
+  # def new
+  #   @comment = Post.new(user_id: current_user.id, movie_id: params[:user_id])
+  # end
+
+  def create
+    @comment = Comment.new(comment_params)
+    @movie = @comment.movie
+    @comments = @movie.comments
+    if @comment.save
+      activity_logger('comment', @movie)
+      redirect_to movie_path(@movie)
+    else
+      @errors = ["Comment cannot be empty"]
+      render 'movies/show'
+    end
+  end
+
+  def edit
+    @comment = Comment.find_by(id: params[:id])
+    @movie =@comment.movie
+    @comments = @movie.comments
+    @conversation = Conversation.new
+    render 'movies/show'
+  end
+
+  def update
+    @comment = Comment.find(params[:id])
+    @movie =@comment.movie
+    @comments = @movie.comments
+    @conversation = Conversation.new
```

Comment:
Also be aware: these variables should be instance variables (variables that start with `@`) _only_ if they are being utilized by another part of your code (e.g. the `render` call below). Remember: `redirect` cannot receive any state, since `redirect` is initiating a completely new HTTP request/response cycle. Thus, if any of these instance variables are not being utilized by the `render` call, they should be changed to local variables.